pkg/apis/deployment/v1alpha/bootstrap.go

```diff
@@ -29,46 +29,66 @@ const (
 	UserNameRoot = "root"
 )
 
+// PasswordSecretName contains user password secret name
+type PasswordSecretName string
+
 const (
 	// PasswordSecretNameNone is magic value for no action
-	PasswordSecretNameNone = "None"
+	PasswordSecretNameNone PasswordSecretName = "None"
 	// PasswordSecretNameAuto is magic value for autogenerate name
-	PasswordSecretNameAuto = "Auto"
+	PasswordSecretNameAuto PasswordSecretName = "Auto"
 )
 
 // PasswordSecretNameList is a map from username to secretnames
-type PasswordSecretNameList map[string]string
+type PasswordSecretNameList map[string]PasswordSecretName
 
 // BootstrapSpec contains information for cluster bootstrapping
 type BootstrapSpec struct {
 	// PasswordSecretNames contains a map of username to password-secret-name
 	PasswordSecretNames PasswordSecretNameList `json:"passwordSecretNames,omitempty"`
 }
 
+// IsNone returns ture if p is empty or None
+func (p PasswordSecretName) IsNone() bool {
+	return p == PasswordSecretNameNone || p == ""
+}
+
+// IsAuto returns ture if p is Auto
+func (p PasswordSecretName) IsAuto() bool {
+	return p == PasswordSecretNameAuto
+}
+
+// Validate validates the password secret name
+func (p PasswordSecretName) Validate() error {
+	if !p.IsNone() {
+		if err := k8sutil.ValidateResourceName(string(p)); err != nil {
+			return maskAny(err)
+		}
+	}
+
+	return nil
+}
+
 // GetSecretName returns the secret name given by the specs. Or None if not set.
-// Except for root user the default is Auto.
-func (s PasswordSecretNameList) GetSecretName(user string) string {
+func (s PasswordSecretNameList) GetSecretName(user string) PasswordSecretName {
 	if s != nil {
-		if password, ok := s[user]; ok {
-			return password
+		if secretname, ok := s[user]; ok {
+			return secretname
 		}
 	}
-	return ""
+	return PasswordSecretNameNone
 }
 
 // getSecretNameForUserPassword returns the default secret name for the given user
-func getSecretNameForUserPassword(deploymentname, username string) string {
-	return deploymentname + "-" + username + "-password"
+func getSecretNameForUserPassword(deploymentname, username string) PasswordSecretName {
+	return PasswordSecretName(deploymentname + "-" + username + "-password")
 }
 
 // Validate the specification.
 func (b *BootstrapSpec) Validate() error {
 	for _, secretname := range b.PasswordSecretNames {
-		if secretname == PasswordSecretNameNone {
-			continue
-		}
-		if err := k8sutil.ValidateResourceName(secretname); err != nil {
-			return maskAny(err)
+		if err := secretname.Validate(); err != nil {
```

Comment:
Should None be allowed for anything other than the `root` user?
Don't think it makes a lot of sense for others.